lib/pages/add_bridge/add_bridge.dart

```diff
@@ -288,31 +300,72 @@ class BotController extends State<AddBridge> {
     });
   }
 
-  /// Handle errors for a social network
-  void _handleError(SocialNetwork socialNetwork) {
+  /// Error handling method with a default error type
+  void _handleError(SocialNetwork socialNetwork,
+      [ConnectionError error = ConnectionError.unknown, String? lastMessage]) {
     setState(() {
       socialNetwork.setError(true);
     });
+
+    String errorMessage;
+
+    switch (error) {
+      case ConnectionError.roomNotFound:
+        errorMessage = 'Room not found';
+        break;
+      case ConnectionError.directChatCreationFailed:
+        errorMessage = 'Failed to create direct chat';
+        break;
+      case ConnectionError.messageSendingFailed:
+        errorMessage = 'Failed to send message';
+        break;
+      case ConnectionError.timeout:
+        errorMessage = 'Operation timed out';
+        break;
+      case ConnectionError.unknown:
+      default:
+        errorMessage = 'An unknown error occurred';
+        break;
+    }
+
+    Logs().v(errorMessage);
+
+    if (lastMessage != null) {
+      showCatchErrorDialog(
+          context, "${L10n.of(context)!.errorSendUsProblem} $lastMessage");
+    } else {
+      showCatchErrorDialog(context,
+          "${L10n.of(context)!.errorConnectionText}.\n\n${L10n.of(context)!.errorSendUsProblem} $errorMessage");
+    }
   }
 
   /// Disconnect from a social network
-  Future<String> disconnectFromNetwork(BuildContext context,
+  Future<ConnectionStatus> disconnectFromNetwork(BuildContext context,
       SocialNetwork network, ConnectionStateModel connectionState) async {
     final String botUserId = '${network.chatBot}$hostname';
+    final SocialNetworkEnum? networkEnum = getSocialNetworkEnum(network.name);
 
     Future.microtask(() {
       connectionState
           .updateConnectionTitle(L10n.of(context)!.loadingDisconnectionDemand);
     });
 
-    final Map<String, RegExp> patterns = getLogoutNetworkPatterns(network.name);
+    final Map<String, RegExp> patterns = getLogoutNetworkPatterns(networkEnum!);
     final String eventName = _getEventName(network.name);
 
     final String? directChat = await _getOrCreateDirectChat(botUserId);
+    if (directChat == null) {
+      _handleError(network, ConnectionError.directChatCreationFailed);
+      return ConnectionStatus.error;
+    }
 
-    final Room? roomBot = client.getRoomById(directChat!);
+    final Room? roomBot = client.getRoomById(directChat);
+    if (roomBot == null) {
+      _handleError(network, ConnectionError.roomNotFound);
+      return ConnectionStatus.error;
+    }
```

Comment:
Shouldn't errors be thrown rather than returned ?

This comment applies to multiple similar code snippets